src/python/tests/test_orthology_benchmark.py

```diff
@@ -0,0 +1,225 @@
+
+    """
+
+    core_dbs = {} # type: Dict
+
+    # autouse=True makes this fixture be executed before any test_* method of this class, and scope='class' to
+    # execute it only once per class parametrization
+    @pytest.fixture(scope='class', autouse=True)
+    def setup(self, multi_dbs: Dict) -> None:
+        """Loads the required fixtures and values as class attributes.
+
+        Args:
+            multi_dbs: Dictionary of unit test databases (fixture).
+
+        """
+        type(self).core_dbs = multi_dbs
+
+    @pytest.mark.parametrize(
+        "species_list, species_set_name, expectation",
+        [
+            (["gallus_gallus", "homo_sapiens"], "default", does_not_raise()),
+            (["homo_sapiens", "zea_mays"], "default", raises(FileExistsError)),
+            (["felis_catus", "zea_mays"], "test", raises(RuntimeError,
+                    match=r"No cores found for the species set 'test' on the specified host."))
+        ]
+    )
+    def test_dump_genomes(self, species_list: List[str], species_set_name: str,
+                          tmp_dir: Path, expectation: ContextManager) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` function
+        when connection to the server host and port can be established.
+
+        Args:
+            species_list: A list of species (genome names).
+            species_set_name: Species set (collection) name.
+            tmp_dir: Unit test temp directory (fixture).
+            expectation: Context manager for the expected exception, i.e. the test will only pass if that
+                    exception is raised. Use :class:`~contextlib.nullcontext` if no exception is expected.
+
+        """
+        with expectation:
+            host = self.core_dbs["gallus_gallus_core_99_6"].dbc.host
+            port = self.core_dbs["gallus_gallus_core_99_6"].dbc.port
+            # user "travis" hardcoded until we find a better solution
+            orthology_benchmark.dump_genomes(species_list, species_set_name, host, port, "travis", tmp_dir)
+
+            out_files = tmp_dir / species_set_name
+            # pylint: disable-next=no-member
+            exp_out = pytest.files_dir / "orth_benchmark"  # type: ignore[attr-defined]
+            for db_name in self.core_dbs:
+                assert file_cmp(Path(out_files, os.environ['USER'] + "_" + db_name + ".fasta"),
+                                Path(exp_out, db_name + ".fasta"))
+
+    def test_dump_genomes_fake_connection(self, tmp_dir: Path) -> None:
+        """Tests :func:`orthology_benchmark.dump_genomes()` function
+        when provided fake server connection details.
+
+        Args:
+            tmp_dir: Unit test temp directory (fixture).
+
+        """
+        with raises(sqlalchemy.exc.OperationalError):
+            orthology_benchmark.dump_genomes(["mus_musculus", "naja_naja"], "default",
+                                             "fake-host", 666, "compara", tmp_dir)
```

Comment:
The "fake" port number used here is an actual reserved TCP/UDP port number. For an "unreal" port number, I'd suggest `65536`.

Reply:
Changed it at both places 😄  Thank you 👍